packages/@lwc/engine-core/src/framework/__tests__/context.spec.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { describe, it, expect, vi, beforeAll, afterEach, afterAll } from 'vitest';
+import { setTrustedContextSet, setContextKeys } from '@lwc/shared';
+import { logWarnOnce } from '../../shared/logger';
+import { connectContext, disconnectContext } from '../modules/context';
+
+// Mock the logger to inspect console output during tests
+vi.mock('../../shared/logger', () => ({
+    logWarnOnce: vi.fn(),
+}));
+
+// Create mock component with a regular, non-contextful property
+const mockComponent = {};
+Object.setPrototypeOf(mockComponent, {
+    regularProp: 'not contextful',
+});
+
+// Create mock renderer
+const mockRenderer = {
+    registerContextProvider: vi.fn(),
+};
+
+// Create mock VM
+const mockVM = {
+    component: mockComponent,
+    elm: null,
+    renderer: mockRenderer,
+} as any;
+
+if (!(globalThis as any).lwcRuntimeFlags) {
+    Object.defineProperty(globalThis, 'lwcRuntimeFlags', { value: {} });
+}
+
+/**
+ * Need to be able to set and reset the flags at will (lwc/features doesn't provide this)
+ */
+const setFeatureFlag = (name: string, value: boolean) => {
+    (globalThis as any).lwcRuntimeFlags[name] = value;
+};
```

Comment:
You can re-use the [setFeatureFlagsForTest](https://github.com/salesforce/lwc/blob/16048aac65baae6acd66240b70c82dd29b468ef6/packages/%40lwc/integration-karma/test/signal/reactivity/index.spec.js#L24) helper instead of defining your own.